src/deepsparse/yolo/pipelines.py

```diff
@@ -243,12 +242,14 @@ def process_engine_outputs(
             batch_predictions.append(image_output.tolist())
             batch_boxes.append(image_output[:, 0:4].tolist())
             batch_scores.append(image_output[:, 4].tolist())
-            batch_labels.append(
-                [
-                    self.class_names[str(class_ids)]
-                    for class_ids in image_output[:, 5].astype(int)
-                ]
-            )
+            batch_labels.append(image_output[:, 5].tolist())
+            if self.class_names is not None:
+                batch_labels[-1] = list(
+                    map(
+                        self.class_names.__getitem__,
+                        list(map(str, list(map(int, batch_labels[-1])))),
+                    )
+                )
```

Comment:
This seems overly complicated, can be made simpler by chaining, also it's bad practice to invoke `dunder` methods directly

```suggestion
                batch_labels[-1] = [
                    getattr(self.class_names, str(int(label)))
                    for label in batch_labels[-1]
                ]
```